plugins/event_engine_plugin/event_engine_plugin.cpp

```diff
@@ -85,14 +89,22 @@ class event_engine_plugin_impl {
 
     bool is_handled_contract(const account_name n) const;
 
+private:
     template<typename Msg>
-    void send_message(const Msg& msg) {
+    void send_message(Msg& msg, uint32_t id) {
         if(dumpstream_opened) {
+            msg.msg_id = id;
```

Comment:
Why `msg.msg_id` initialized here? It will be more correct it in the Message constructor.